metis-dereference/metis-dereference-import/src/main/java/eu/europeana/metis/dereference/vocimport/utils/DereferenceValidationUtils.java

```diff
@@ -0,0 +1,64 @@
+package eu.europeana.metis.dereference.vocimport.utils;
+
+import edu.emory.mathcs.backport.java.util.Arrays;
+import eu.europeana.metis.dereference.vocimport.exception.VocabularyImportException;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Utils class focused on validating values used in dereferencing
+ */
+public class DereferenceValidationUtils {
+
+    private static final String PROPERTY_FIELD_NAME = "valid.directories.values";
+    private String dereferencingPropertiesLocation =
+            "metis-dereference/metis-dereference-rest/src/main/resources/dereferencing.properties";
+
+    private final List<String> validDirectoriesValues;
+
+    /**
+     * Constructor of Utils class
+     * @throws VocabularyImportException if an issue occurs while initializing
+     */
+    public DereferenceValidationUtils() throws VocabularyImportException {
+        validDirectoriesValues = Arrays.asList(readProperty());
+    }
+    // This constructor is used for testing purposes
+    DereferenceValidationUtils(String dereferencingPropertiesLocation) throws VocabularyImportException {
+        this.dereferencingPropertiesLocation = dereferencingPropertiesLocation;
+        validDirectoriesValues = Arrays.asList(readProperty());
+    }
+
+    /**
+     * Method that verifies if the given directory is valid
+     * @param directoryToEvaluate The directory to evaluate as a string
+     * @return True if is valid; False otherwise
+     */
+    public boolean isDirectoryValid(String directoryToEvaluate) {
+        if (CollectionUtils.isEmpty(validDirectoriesValues)) {
+            return false;
```

Comment:
Try to reduce returns by having a variable and then returning that once at the end.

Reply:
Done